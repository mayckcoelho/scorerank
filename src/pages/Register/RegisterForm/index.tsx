import React, { useRef } from 'react';
import { Form } from '@unform/mobile';
import { SubmitHandler, FormHandles } from '@unform/core';
import * as Yup from "yup";

import { useAuth } from '../../../contexts/auth';
import Input from "../../../components/Input";
import Buttom from "../../../components/Buttom";
import IRegisterRequest from "../../../interfaces/IRegisterRequest";
interface IDictionary {
    [index: string]: string;
}

const RegisterForm: React.FC = () => {
    const { register } = useAuth();
    const formRef = useRef<FormHandles>(null);

    const handleSubmit: SubmitHandler<IRegisterRequest> = async (data) => {
        try {
            const schema = Yup.object().shape({
                name: Yup.string().required("Nome obrigatório"),
                email: Yup.string().email("Informe um e-mail válido").required("E-mail obrigatório"),
                password: Yup.string().min(6, "Informe ao menos 6 caractéres").required("Senha obrigatória"),
                confirm_password: Yup.string().test('passwords-match', 'Senhas não são iguais', function (value) {
                    return this.parent.password === value;
                }),
            });

            await schema.validate(data, {
                abortEarly: false
            });

            register(data);

        } catch (err) {
            const validationErrors = {} as IDictionary;
            if (err instanceof Yup.ValidationError) {
                err.inner.forEach(error => {
                    validationErrors[error.path] = error.message;
                });
                formRef.current?.setErrors(validationErrors);
            }
        }
    };

    return (
        <Form ref={formRef} onSubmit={handleSubmit}>
            <Input name="firstname"
                returnKeyType="next"
                onSubmitEditing={() => {
                    const emailInput = formRef.current?.getFieldRef('email');
                    emailInput.focus();
                }}
                blurOnSubmit={false}
                icon="user"
                placeholder="Nome" />
            <Input name="email"
                returnKeyType="next"
                onSubmitEditing={() => {
                    const passwordInput = formRef.current?.getFieldRef('password');
                    passwordInput.focus();
                }}
                blurOnSubmit={false}
                icon="envelope-o"
                placeholder="Endereço de Email" />
            <Input name="password"
                returnKeyType="next"
                onSubmitEditing={() => {
                    const confirmInput = formRef.current?.getFieldRef('confirm_password');
                    confirmInput.focus();
                }}
                icon="key"
                placeholder="Senha"
                secureTextEntry />
            <Input name="confirm_password" icon="key" placeholder="Confirmar Senha" secureTextEntry />
            <Buttom text="Registrar" color="#215760" onPress={() => formRef.current?.submitForm()} />
        </Form>
    )
}

export default RegisterForm;