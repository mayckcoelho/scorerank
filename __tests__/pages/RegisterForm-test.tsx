import React from 'react';
import RegisterForm from '../../src/pages/Register/RegisterForm';
import renderer from 'react-test-renderer';

describe('RegisterForm should render correctly', () => {
    it('match snapshot', () => {
        const registerform = renderer
            .create(<RegisterForm />)
            .toJSON();
        expect(registerform).toMatchSnapshot();
    });
})