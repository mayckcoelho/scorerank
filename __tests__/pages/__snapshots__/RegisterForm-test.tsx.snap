// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterForm should render correctly match snapshot 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "marginBottom": 15,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRadius": 4,
            "color": "#444",
            "flexDirection": "row",
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#b5c0ce",
              "fontSize": 16,
            },
            undefined,
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <TextInput
        allowFontScaling={true}
        blurOnSubmit={false}
        keyboardAppearance="dark"
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        placeholder="Nome"
        placeholderTextColor="#b5c0ce"
        rejectResponderTermination={true}
        returnKeyType="next"
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "fontSize": 15,
              "marginLeft": 5,
            },
          ]
        }
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "marginBottom": 15,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRadius": 4,
            "color": "#444",
            "flexDirection": "row",
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#b5c0ce",
              "fontSize": 16,
            },
            undefined,
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <TextInput
        allowFontScaling={true}
        blurOnSubmit={false}
        keyboardAppearance="dark"
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        placeholder="Endereço de Email"
        placeholderTextColor="#b5c0ce"
        rejectResponderTermination={true}
        returnKeyType="next"
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "fontSize": 15,
              "marginLeft": 5,
            },
          ]
        }
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "marginBottom": 15,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRadius": 4,
            "color": "#444",
            "flexDirection": "row",
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#b5c0ce",
              "fontSize": 16,
            },
            undefined,
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <TextInput
        allowFontScaling={true}
        keyboardAppearance="dark"
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        placeholder="Senha"
        placeholderTextColor="#b5c0ce"
        rejectResponderTermination={true}
        returnKeyType="next"
        secureTextEntry={true}
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "fontSize": 15,
              "marginLeft": 5,
            },
          ]
        }
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "marginBottom": 15,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRadius": 4,
            "color": "#444",
            "flexDirection": "row",
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#b5c0ce",
              "fontSize": 16,
            },
            undefined,
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <TextInput
        allowFontScaling={true}
        keyboardAppearance="dark"
        onChangeText={[Function]}
        placeholder="Confirmar Senha"
        placeholderTextColor="#b5c0ce"
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "fontSize": 15,
              "marginLeft": 5,
            },
          ]
        }
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#215760",
        "borderColor": "black",
        "borderRadius": 4,
        "borderStyle": "solid",
        "borderWidth": 0,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 10,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 40,
        "opacity": 1,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 15,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Registrar
    </Text>
  </View>
</View>
`;
